unity-renderer/Assets/Scripts/MainScripts/DCL/AvatarSystem/Loader/FacialFeatureRetriever.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Threading;
+using Cysharp.Threading.Tasks;
+using DCL;
+using UnityEngine;
+
+namespace AvatarSystem
+{
+    public class FacialFeatureRetriever : IFacialFeatureRetriever
+    {
+        AssetPromise_Texture mainTexturePromise = null;
+        AssetPromise_Texture maskTexturePromise = null;
+
+        public async UniTask<(Texture main, Texture mask)> Retrieve( string mainTextureUrl, string maskTextureUrl, CancellationToken ct = default)
+        {
+            if (ct.IsCancellationRequested)
+                throw new OperationCanceledException();
+
+            try
+            {
+                AssetPromiseKeeper_Texture.i.Forget(mainTexturePromise);
```

Comment:
anyway you can inject this singleton instance from constructor? it would make it scalable for tests